lib/spack/spack/test/test_suite.py

```diff
@@ -307,6 +307,7 @@ def test_test_part_fail(tmpdir, install_mockery, mock_fetch, mock_test_stage):
         assert status == TestStatus.FAILED
 
 
+@pytest.mark.not_on_windows("echo not available on Windows")
```

Comment:
Echo in the sense that you can run `which` on it is not available as it is an internal command to `cmd`, I think we can still reasonably find something that will echo to stdout, either using Python or composing our use of echo to be a batch script.

Reply:
@johnwparent Do you want to mark this as resolved, or do you have a suggestion here?

Reply:
I've resolved it, this issue is more pervasive than just this one instance and the solution is outside the scope of this PR and should/will be done in a follow up. 

Thanks for pointing out this is unresolved @wdconinc!